tests/unit/test_hnswlib.cpp

```diff
@@ -54,19 +56,129 @@ TEST_F(HNSWLibTest, resizeIndex) {
     ASSERT_EQ(VecSimIndex_IndexSize(index), 0);
 
     float a[dim];
+
+    // add up to n
+    for (size_t i = 0; i < n; i++) {
+        for (size_t j = 0; j < dim; j++) {
+            a[j] = (float)i;
+        }
+        VecSimIndex_AddVector(index, (const void *)a, i);
+    }
+    // the size and the capacity should be equal
+    ASSERT_EQ(reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexCapacity(),
+              VecSimIndex_IndexSize(index));
+    // the capcity shouldnt be changed
+    ASSERT_EQ(reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexCapacity(), n);
+
+    // add another vector to exceed the initial capacity
+    VecSimIndex_AddVector(index, (const void *)a, n);
+
+    // the capacity should be now aligned with the block size
+    ASSERT_EQ(reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexCapacity() % bs, 0);
+    // new capacity = intiailcapacity + blockSize - intiailcapacity % blockSize
+    ASSERT_EQ(reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexCapacity(),
+              n + bs - n % bs);
+
+    VecSimIndex_Free(index);
+}
+
+// case 1: intiailCapacity much larger than block size, and it is not aligned
+TEST_F(HNSWLibTest, resizeNAlignIndex_initailCapacity_is_big) {
+    size_t dim = 4;
+    size_t n = 32;
+    size_t bs = 3;
+    VecSimParams params{.algo = VecSimAlgo_HNSWLIB,
+                        .hnswParams = HNSWParams{.type = VecSimType_FLOAT32,
+                                                 .dim = dim,
+                                                 .metric = VecSimMetric_L2,
+                                                 .initialCapacity = n,
+                                                 .blockSize = bs}};
+    VecSimIndex *index = VecSimIndex_New(&params);
+    ASSERT_EQ(VecSimIndex_IndexSize(index), 0);
+
+    float a[dim];
+
+    // add up to blocksize + 1
+    for (size_t i = 0; i < bs + 1; i++) {
+        for (size_t j = 0; j < dim; j++) {
+            a[j] = (float)i;
+        }
+        VecSimIndex_AddVector(index, (const void *)a, i);
+    }
+
+    // the capcity shouldnt be changed
+    ASSERT_EQ(reinterpret_cast<HNSWIndex *>(index)->getHNSWIndex()->getIndexCapacity(), n);
+
+    // delete random vector, to get size % block_size == 0
+    VecSimIndex_DeleteVector(index, 1);
```

Comment:
no. and thats the wanted behaviour

Reply:
I understand. Do you have a test case for this scenario (deleting a vector when there is a block to remove, but avoiding it because `size % block_size != 0`). If not, consider adding it.